### PR TITLE
Pulp: add cron job that deletes container registries

### PR DESCRIPTION
**Pull request by jluk-codi**

### CODE REVIEW

File: modules/opencontrail_ci/manifests/logserver.pp [hunk 1]
```diff
@@ -105,7 +105,7 @@
 
   file { '/etc/logrotate.d/zuul-jobs-stats':
     ensure  => file,
-    content => 'pupppet:///modules/opencontrail_ci/zuul-jobs-stats/logrotate',
+    content => 'puppet:///modules/opencontrail_ci/zuul-jobs-stats/logrotate',
```
Comment (wurbanski) [inline modules/opencontrail_ci/manifests/logserver.pp:108]: :+1:

File: modules/opencontrail_ci/files/pulp/artifact_curator.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                if subhead['id']:
+                    change, patchset = subhead['id'].split(',')
+                    running_buildsets.append({'change': change, 'patchset': patchset, 'buildset': buildset_id})
+
+# Delete all registries created by non-running buildsets
+to_delete = []
+print()
+for cont in registries:
+    if cont in running_buildsets:
+        print('Not removing', cont)
```
Comment (kklimonda) [inline modules/opencontrail_ci/files/pulp/artifact_curator.py]: Change all print into logging statements and log to /var/log/{something}.log

File: modules/opencontrail_ci/files/pulp/artifact_curator.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    else:
+        print('Removing,', cont)
+        to_delete.append(cont)
+for cont in to_delete:
+    name = 'registry_{change}_{patchset}_{buildset}'.format(**cont)
+    print(name)
+    if not dry_run:
+        # TODO handle already removed containers
```
Comment (kklimonda) [inline modules/opencontrail_ci/files/pulp/artifact_curator.py]: Is it going to fail when container has already been deleted?

File: modules/opencontrail_ci/files/pulp/artifact_curator.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+dry_run = False
+
+registries = []
+running_buildsets = []
+
+# Get the list of running containers first to prevent race condition
+cli = Client()
+containers = cli.containers()
+for container in containers:
+    #print(container['Names'])
```
Comment (kklimonda) [inline modules/opencontrail_ci/files/pulp/artifact_curator.py]: Remove commented out code

**Review by kklimonda — CHANGES_REQUESTED** [review]

**Review by kklimonda — APPROVED** [review]

**Pull request merged**